compiler-rt/lib/hwasan/hwasan_allocator.cpp

```diff
@@ -289,6 +289,9 @@ static void HwasanDeallocate(StackTrace *stack, void *tagged_ptr) {
   CHECK(tagged_ptr);
   void *untagged_ptr = UntagPtr(tagged_ptr);
 
+  if (RunFreeHooks(tagged_ptr))
```

Comment:
why do you need to move?
I guess invalid free is still invalid free,  regardless of the new tool?

Reply:
A concrete use case: The program is doing a free without the correct tag, and you want to silence the invalid-free check. In your runtime library, you can declare the hook, ignore the free(), then grab the tag and put it in the pointer, then call free again.

I figure it's more feature-rich for no downsides :).